src/mapping/query/function/ceil.rs

```diff
@@ -0,0 +1,117 @@
+use super::prelude::*;
+use super::util::round_to_precision;
+
+#[derive(Debug)]
+pub(in crate::mapping) struct CeilFn {
+    query: Box<dyn Function>,
+    precision: Option<Box<dyn Function>>,
+}
+
+impl CeilFn {
+    #[cfg(test)]
+    pub(in crate::mapping) fn new(
+        query: Box<dyn Function>,
+        precision: Option<Box<dyn Function>>,
+    ) -> Self {
+        Self { query, precision }
+    }
+}
+
+impl Function for CeilFn {
+    fn execute(&self, ctx: &Event) -> Result<Value> {
+        let precision = optional!(ctx, self.precision, Value::Integer(v) => v).unwrap_or(0);
+        let res = required!(ctx, self.query,
+                            Value::Float(f) => {
+                                Value::Float(round_to_precision(f, precision, f64::ceil))
+                            },
+                            v@Value::Integer(_) => v
```

Comment:
What does this `v@Value` syntax do? I was expecting this to be `Value::Integer(v) => v`.

Reply:
Never mind, I see https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html#-bindings . I was not aware of that syntax :smile:

Reply:
In `v@Value` the `v` is bound to the whole pattern - so in this case, for example `Value::Integer(42)`. 

With `Value::Integer(v) => v`, `v` would be the wrapped value, so, just the `42`.

Since that branch is returning a `Value`, it would have to do `Value::Integer(v) => Value::Integer(v)`, ie. leave the value unchanged. So instead, it just returns the original value, `v`.